### ISSUE TITLE
Memory leak in CgInfluence

### ISSUE THREAD

**Issue by schroedk** [aai-institute/pyDVL#497]
Working with a Resnet50 modek on imagenet revealed a memory issue with the implementation of [CgInfluence](https://github.com/aai-institute/pyDVL/blob/develop/src/pydvl/influence/torch/influence_function_model.py#L425).

The issue is due to precomputing the averaged gradient over the training dataset, while keeping the computation graph for computing the second order derivatives. This is not feasible for large datasets.

**Resolution**:

Expose the parameter `precopmute_grad`. Setting to False, results in the averaging of the second order derivatives over the batches, which requires less memory (no need to keep the computation graph) but is slower in general.

 The same applies to [ArnoldiInfluence](https://github.com/aai-institute/pyDVL/blob/develop/src/pydvl/influence/torch/influence_function_model.py#L717).
